src/components/Modal/Modal.jsx

```diff
@@ -41,6 +42,17 @@ class Modal extends React.Component {
     if (this.props.clickToClose) {
       document.addEventListener('click', this.handleDocumentClicked, true);
     }
+    // Conditionally track the event.
+    if (this.props.enableAnalytics) {
+      dispatchAnalyticsEvent({
+        type: 'Modal',
+        action: 'show',
+        status: this.props.status,
+        title: this.props.title,
+        primaryButtonText: this.props.primaryButton?.text,
+        secondaryButtonText: this.props.secondaryButton?.text,
```

Comment:
Like we talked about in the meeting (minutes ago from when I'm writing this :smile: ), can we group these together in a nested object. Maybe call it `details` or `context` or something. This will collect all the event-specific details, which are expected to be very different between components and events.

Reply:
Done.  I like `details` since context has a lot of other meanings.